apps/condo/domains/billing/utils/countrySpecificQRCodeParsers/ru.qrCodeParser.js

```diff
@@ -17,21 +17,20 @@ const { get } = require('lodash')
  * @return {TRUQRCodeFields}
  */
 function parseRUReceiptQRCode (qrStr) {
-    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
+    const buffer = Buffer.from(qrStr, 'base64')
+    const { encoding: detectedEncoding } = jschardet.detect(buffer)
 
-    if (!matches) {
-        throw new Error('Invalid QR code')
-    }
+    // In Russia, only two formats are used: utf and windows-1251. Therefore, here we strictly control the formats
+    const encoding = detectedEncoding && detectedEncoding.toUpperCase() === 'UTF-8' ? 'UTF-8' : 'WINDOWS-1251'
 
-    const requisitesStr = get(matches, ['groups', 'requisitesStr'], '')
+    // Skip decoding for utf string
+    const decodedRequisitesStr = encoding === 'UTF-8' ? buffer.toString() : iconv.decode(buffer, encoding)
 
-    // https://encoding.spec.whatwg.org/#koi8-r
-    // https://encoding.spec.whatwg.org/#windows-1251
-    const encodingTag = get(matches, ['groups', 'encodingTag'])
-    const encoding = get({ 1: 'cp1251', 2: 'utf-8', 3: 'koi8-r' }, encodingTag, 'utf-8')
+    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(decodedRequisitesStr)
 
-    // Skip decoding for utf string
-    const decodedRequisitesStr = encodingTag === '2' ? requisitesStr : iconv.decode(requisitesStr, encoding)
+    if (!matches) {
+        throw new Error('Invalid QR code')
+    }
 
     return Object.fromEntries(decodedRequisitesStr.split('|').map((part) => part.split('=', 2)))
```

Comment:
Why do you parse here the full string including header (ST...)? Please use only requisites part (requisitesStr)